### PR TITLE
[config] Document `bugprone-assignment-in-if-condition` and `misc-const-correctness`

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: config/labels/analyzers/clang-tidy.json [hunk 1]
```diff
@@ -138,6 +138,13 @@
       "sei-cert:pre31-c",
       "severity:MEDIUM"
     ],
+    " bugprone-assignment-in-if-condition": [
```
Comment (bruntib) [inline config/labels/analyzers/clang-tidy.json:141]: Extra space in the checker name.

File: config/labels/analyzers/clang-tidy.json [hunk 2]
```diff
@@ -5085,6 +5092,12 @@
       "profile:sensitive",
       "severity:LOW"
     ],
+    "misc-const-correctness": [
+      "doc_url:https://clang.llvm.org/extra/clang-tidy/checks/misc/const-correctness.html",
+      "profile:extreme",
```
Comment (whisperity) [inline config/labels/analyzers/clang-tidy.json:5097]: I'm not sure whether we should add this to any profile by default, but the severity looks okay.

**Comment by sylvestre** [comment]
As you wish :)
I don't have any opinion on this!

File: config/labels/analyzers/clang-tidy.json (hunk 2 above)
Comment (dkrupp) [reply config/labels/analyzers/clang-tidy.json:5097]: I think it is okay in the extreme profile

**Review by dkrupp — APPROVED** [review]
LGTM. thanks for the patch.

**Pull request merged**